### PR TITLE
Inner/Outer Stairs translation in external mods

### LINKED ISSUE #2582: Inner and Outer Stairs translation is broken in external mods in non-english languages
In [stairs module](https://github.com/minetest-game-mods/stairs): 
Since in `init.lua` at lines 488 and 490, registration is done as
```lua
stairs.register_stair_inner(subname, recipeitem, groups, images, "",
    sounds, worldaligntex, S("Inner " .. desc_stair))
stairs.register_stair_inner(subname, recipeitem, groups, images, "",
    sounds, worldaligntex, S("Outer " .. desc_stair))
```
instead of
```lua
stairs.register_stair_inner(subname, recipeitem, groups, images, "",
    sounds, worldaligntex, S("Inner @1 Stair", desc_stair))
stairs.register_stair_inner(subname, recipeitem, groups, images, "",
    sounds, worldaligntex, S("Outer @1 Stair", desc_stair))
```
External mods cannot have Inner and Outer Stair rendered correctly when using this function.
As example, nether mod use `stairs.register_stair_and_slab` with `"Nether Stair"` as `desc_stair` parameter, and with translation support in minetest-mods/nether/pull/15, will use it with `S(Nether Stair")`.

In french, this result in `Inner Escalier du nether` and `Outer Escalier du nether` instead of `Escalier intérieur du Nether` and `Escalier extérieur du Nether`.

Since `S("Inner" .. desc_stair)` cannot take translations from external mods locales files, this cannot be solved without changing the format of the `desc_stair` which would be like `Wooden` or `Nether` instead of `Wooden Stair` or `Nether Stair` would and this requires changing all translations in stairs mod’s locales, but also breaking api in some kind: for the moment, this works well for english-only mods, but this change would result in external mods using this function having twice the word Stair in resulted descriptions of stairs they have defined.

Considering all the above, should we do this change ?

### PR DESCRIPTION
**Pull request by louisroyer**
Closes #2582

EDIT: Actually implements https://github.com/minetest/minetest_game/issues/2582#issuecomment-582447500

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -468,13 +468,13 @@ end
 -- Nodes will be called stairs:{stair,slab}_<subname>
 
 function stairs.register_stair_and_slab(subname, recipeitem, groups, images,
-		desc_stair, desc_slab, sounds, worldaligntex)
+		desc_stair, desc_slab, sounds, worldaligntex, full_desc_stair_outer, full_desc_stair_inner)
```
Comment (sfan5) [inline mods/stairs/init.lua]: Line too long (80 column limit)
I also suggest calling these `desc_stair_outer`,`desc_stair_inner` because that's shorter.

**Comment by sfan5** [comment]
The documentation in `game_api.txt` also needs to be updated.

File: game_api.txt [hunk 2]
```diff
@@ -752,12 +752,18 @@ delivered with Minetest Game, to keep them compatible with other mods.
...
+```
+stairs.register_stair_and_slab(subname, recipeitem, groups, images, desc_stair, desc_slab,
+	sounds, worldaligntex, desc_stair_outer, desc_stair_inner)
+```
 
  * A wrapper for stairs.register_stair, stairs.register_slab, stairs.register_stair_inner, stairs.register_stair_outer
  * Uses almost the same arguments as stairs.register_stair
- * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed
+ * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed unless
+ 	you set arguments `desc_stair_outer` or `desc_stair_inner`
```
Comment (sfan5) [inline game_api.txt]:
```suggestion
 	`desc_stair_outer` or `desc_stair_inner` are specified, which are used instead.
```

**Comment by louisroyer** [comment]
> Looks good, hope you've tested it.
I just re-tested and it works.

**Comment by sfan5** [comment]
@minetest/game bump?

**Comment by dacmot** [comment]
This is quite painful from non-English point of view. Are there any chances to see this merged for the next MT release? Thanks.

File: game_api.txt [hunk 3]
```diff
@@ -752,12 +752,18 @@ delivered with Minetest Game, to keep them compatible with other mods.
  * `worldaligntex`: A bool to set all textures world-aligned. Default false. See [Tile definition]
  * `full_description`: Overrides the description, bypassing string concatenation. This is useful for translation. (optional)
 
-`stairs.register_stair_and_slab(subname, recipeitem, groups, images, desc_stair, desc_slab, sounds, worldaligntex)`
+```
+stairs.register_stair_and_slab(subname, recipeitem, groups, images, desc_stair, desc_slab,
+	sounds, worldaligntex, desc_stair_outer, desc_stair_inner)
```
Comment (paramat) [inline game_api.txt:757]: Reorder to: inner, outer

File: game_api.txt [hunk 4]
```diff
@@ -752,12 +752,18 @@ delivered with Minetest Game, to keep them compatible with other mods.
...
+```
 
  * A wrapper for stairs.register_stair, stairs.register_slab, stairs.register_stair_inner, stairs.register_stair_outer
  * Uses almost the same arguments as stairs.register_stair
- * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed
+ * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed unless
+ 	`desc_stair_outer` or `desc_stair_inner` are specified, which are used instead.
  * `desc_slab`: Description for slab node
+ * `desc_stair_outer`: Description for outer stair node
+ * `desc_stair_inner`: Description for inner stair node
```
Comment (paramat) [inline game_api.txt:766]: Reorder to: inner, outer

File: game_api.txt [hunk 5]
```diff
@@ -752,12 +752,18 @@ delivered with Minetest Game, to keep them compatible with other mods.
...
+```
+stairs.register_stair_and_slab(subname, recipeitem, groups, images, desc_stair, desc_slab,
+	sounds, worldaligntex, desc_stair_outer, desc_stair_inner)
+```
 
  * A wrapper for stairs.register_stair, stairs.register_slab, stairs.register_stair_inner, stairs.register_stair_outer
  * Uses almost the same arguments as stairs.register_stair
- * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed
+ * `desc_stair`: Description for stair nodes. For corner stairs 'Inner' or 'Outer' will be prefixed unless
+ 	`desc_stair_outer` or `desc_stair_inner` are specified, which are used instead.
```
Comment (paramat) [inline game_api.txt:763]: Reorder to: inner, outer

File: mods/stairs/init.lua [hunk 6]
```diff
@@ -468,13 +468,14 @@ end
 -- Nodes will be called stairs:{stair,slab}_<subname>
 
 function stairs.register_stair_and_slab(subname, recipeitem, groups, images,
-		desc_stair, desc_slab, sounds, worldaligntex)
+		desc_stair, desc_slab, sounds, worldaligntex,
+		desc_stair_outer, desc_stair_inner)
```
Comment (paramat) [inline mods/stairs/init.lua:472]: Reorder to: inner, outer

**Comment by paramat** [comment]
Throughout the code and API documentation the order 'inner, outer' has always been used, but you use 'outer, inner'.
Please could you reverse the order? Then i will approve and merge this (have not tested but changes seem ok).

**Comment by paramat** [comment]
:+1:

**Pull request merged**